modules/services/nix-gc.nix

```
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.nix.gc;
+
+  mkCalendarInterval = frequency:
+    let
+      freq = {
+        "hourly" = [{ Minute = 0; }];
+        "weekly" = [{
+          Weekday = 1;
+          Hour = 0;
+          Minute = 0;
+        }];
+        "monthly" = [{
+          Day = 1;
+          Hour = 0;
+          Minute = 0;
+        }];
+        "semiannually" = [
+          {
+            Month = 1;
+            Day = 1;
+            Hour = 0;
+            Minute = 0;
+          }
+          {
+            Month = 7;
+            Day = 1;
+            Hour = 0;
+            Minute = 0;
+          }
+        ];
+        "annually" = [{
+          Month = 1;
+          Day = 1;
+          Hour = 0;
+          Minute = 0;
+        }];
+      };
+    in freq.${frequency};
+  nix-package = if config.nix.enable && config.nix.package != null then
+    config.nix.package
+  else
+    pkgs.nix;
+in {
+  meta.maintainers = [ maintainers.shivaraj-bh ];
+
+  options = {
+    nix.gc = {
+      automatic = mkOption {
+        default = false;
+        type = types.bool;
+        description = "Automatically run the garbage collector at a specific time.";
+      };
+
+      frequency = mkOption {
+        type =
+          types.enum [ "hourly" "weekly" "monthly" "semiannually" "annually" ];
+        description = ''
+          The frequency at which to run the garbage collector.
+
+          These enums are based on special expressions from the
+          {manpage}`systemd.time(7)`
+        '';
+        example = "monthly";
+        default = "weekly";
+      };
+
+      options = mkOption {
+        default = null;
+        example = "--max-freed $((64 * 1024**3))";
+        type = types.nullOr types.str;
+        description = ''
+          Options given to {file}`nix-collect-garbage` when the
+          garbage collector is run automatically.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf (config.nix.enable && cfg.automatic) (mkMerge [
```

Since `nixPackage` is regardless of the value of `config.nix.enable`:

```suggestion
  config = lib.mkIf cfg.automatic (mkMerge [
```